### ISSUE TITLE
Implement DialogPanel?

### ISSUE THREAD

**Issue by ivarlovlie** [rgossiaux/svelte-headlessui#112]
Headless ui has [Dialog.Panel](https://headlessui.com/react/dialog#dialog-panel), I am unable to find it in svelte-headlessui. Is this because it is not implemented here, or am i missing something about the need for Dialog.Panel in this port?

Thanks!

**Comment by karmaral** [comment]
+1 for this

**Comment by extremelygooddeveloper** [comment]
I would be willing to invest time in a PR if there was some kind of CONTRIBUTING-style documentation with pointers on how to port components.

**Comment by extremelygooddeveloper** [comment]
[My first pass](https://github.com/extremelygooddeveloper/svelte-headlessui/) on this seems to do the job, probably not production ready, but would love to make it so in a pr.

**Comment by petr24** [comment]
@extremelygooddeveloper Unless there is a misunderstanding but the dialog panel is on the example page here https://svelte-headlessui.goss.io/docs/dialog

**Comment by extremelygooddeveloper** [comment]
@petr24 Hmm, i only see these 
<img width="218" alt="image" src="https://user-images.githubusercontent.com/38570165/195973470-eef3af1a-f460-4577-8d37-7cf7332cff6a.png">

**Comment by petr24** [comment]
Have no idea what page you're visiting, I go to the link on chrome and this is what you should see...

![Screen Shot 2022-10-18 at 10 59 02 AM](https://user-images.githubusercontent.com/3904464/196508513-3e8610ae-a1aa-4c40-b257-0a9750a2271b.png)

**Comment by extremelygooddeveloper** [comment]
Yes, there is no DialogPanel on that page.

**Comment by petr24** [comment]
Well there is no "DialogPanel" in tailwind css or headless UI, it's called just "Dialog". I think this can be closed out.

**Comment by petr24** [comment]
> Well there is no "DialogPanel" in tailwind css or headless UI, it's called just "Dialog". I think this can be closed out.

So I was mis-understanding your question, and my answers were wrong. I see you were asking about "Dialog.Panel" with svelte variant being "DialogPanel", however I'm not sure it's necessary. I looked at the DOM for both headless UI example, and svelte headless UI,  and it's the same. Parent Div with three children, h3, two divs. So it seems like this package has the function of Dialog.Panel without explicitly copying the headless UI api. 

Svelte Headless UI
![Screen Shot 2022-10-20 at 8 08 55 AM](https://user-images.githubusercontent.com/3904464/196988867-fc78dcc0-c309-49b6-a97f-0ff13cf276c2.png)
 
Headless UI
![Screen Shot 2022-10-20 at 8 14 59 AM](https://user-images.githubusercontent.com/3904464/196988832-c9fbcae5-acde-47f6-9020-6adf3cc768e9.png)

**Comment by coin-au-carre** [comment]
For reference, if it can be useful these layouts uses DialogPanel: https://tailwindui.com/components/application-ui/application-shells/sidebar. 
I have tried to replace DialogPanel with Dialog without good visual for the moment

**Comment by extremelygooddeveloper** [comment]
@coin-au-carre Yes, that was my experience as well, iirc my first pass on implementing DialogPanel worked for me.
Feel free to try it out:
Repo: https://github.com/extremelygooddeveloper/svelte-headlessui
npm: https://www.npmjs.com/package/@developermuch/dev-svelte-headlessui

**Comment by coin-au-carre** [comment]
Thanks @extremelygooddeveloper I will try it :)

**Comment by AaronNBrock** [comment]
Bumping that @iv-ar should turn this into a PR!

**Comment by HemalR** [comment]
> I mis-understood your question, and my answers were wrong. I see you were asking about "Dialog.Panel" with svelte variant being "DialogPanel", however I'm not sure it's necessary. I looked at the DOM for both headless UI example, and svelte headless UI, and it's the same. Parent Div with three children, h3, two divs. So it seems like this package has the function of Dialog.Panel without explicitly copying the headless UI api.
> 


I think the main functionality is to trigger a close of the dialog when a click-event is made outside of it. I would echo the requests for @iv-ar to turn this into a PR. Thank you all in advance!

**Comment by zach-hopkins** [comment]
I'm not really sure what DialogPanel does. I've just replaced them with <divs> and never noticed anything wrong.

**Comment by HemalR** [comment]
> I'm not really sure what DialogPanel does. I've just replaced them with divs and never noticed anything wrong.

Are you able to close the dialog by clicking outside of it?

**Comment by zach-hopkins** [comment]
> > I'm not really sure what DialogPanel does. I've just replaced them with divs and never noticed anything wrong.
> 
> Are you able to close the dialog by clicking outside of it?

Ah yes I believe that may be it. I cannot do that by default, however I do not mind creating that functionality manually.

**Comment by petr24** [comment]
From the documentation, clicking outside modal/dialog works as expected. 
Is this not working for everyone?

```
The close event is fired when an open dialog is dismissed, 
which happens when the user clicks outside of the contents of your dialog or presses the Escape key. 
You can use this callback to set open back to false and close your dialog.

<script>
  import {
    Dialog,
    DialogOverlay,
    DialogTitle,
    DialogDescription,
  } from "@rgossiaux/svelte-headlessui";
  let isOpen = true;
</script>

<!-- Pass `isOpen` to the `open` prop and use the `on:close` handler to set it back to false when the user clicks outside of the dialog or presses the escape key. -->
<Dialog open={isOpen} on:close={() => (isOpen = false)}>
  <DialogOverlay />

  <DialogTitle>Deactivate account</DialogTitle>
  <DialogDescription>
    This will permanently deactivate your account
  </DialogDescription>

  <p>
    Are you sure you want to deactivate your account? All of your data will be
    permanently removed. This action cannot be undone.
  </p>

  <!-- You can render additional buttons to dismiss your dialog by setting `isOpen` to `false`. -->
  <button on:click={() => (isOpen = false)}>Deactivate</button>
  <button on:click={() => (isOpen = false)}>Cancel</button>
</Dialog>

```
